pyiceberg/catalog/__init__.py

```diff
@@ -394,6 +394,11 @@ def table_exists(self, identifier: Union[str, Identifier]) -> bool:
         Returns:
             bool: True if the table exists, False otherwise.
         """
+        try:
```

Comment:
can we get rid of the other `table_exists` function in the same file? 
https://github.com/apache/iceberg-python/blob/87c13b54e44b0c2c5b9cce79072ccdc3d58d6d63/pyiceberg/catalog/__init__.py#L668-L673

Reply:
what do you think about changing the REST catalog's `table_exists` implementation to be similar to this? 
https://github.com/apache/iceberg-python/pull/512/files#diff-3bda7391ebd8aa3dcfd6703d8d2764830b9d9c35fa854188a37d69611274bd3dR722-R727

maybe calls `super().table_exists()`?

Reply:
I think it is better for RestCatalog to maintain a separate implementation to make a head request to [/v1/{prefix}/namespaces/{namespace}/tables/{table}](https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml#L814), ref: https://github.com/apache/iceberg-python/pull/512#discussion_r1519661896, https://github.com/apache/iceberg-python/issues/507#issue-2177108431

The try-catch implementation is for other non-rest catalogs and thus it is put in the `MetastoreCatalog` instead of the `Catalog` interface.